traffic_ops/testing/api/v4/parameters_test.go

```diff
@@ -141,7 +179,78 @@ func UpdateTestParameters(t *testing.T) {
 	if respParameter.Value != expectedParameterValue {
 		t.Errorf("results do not match actual: %s, expected: %s", respParameter.Value, expectedParameterValue)
 	}
+	//update back to old value for safe deletion in pre-requisite
+	alert, _, err = TOSession.UpdateParameter(remoteParameter.ID, old, nil)
+	if err != nil {
+		t.Errorf("cannot UPDATE Parameter by id: %v - %v", err, alert)
+	}
+}
+
+func UpdateParametersInvalidValue(t *testing.T) {
+	firstParameter := testData.Parameters[0]
+	// Retrieve the Parameter by name so we can get the id for the Update
+	params := url.Values{}
+	params.Set("name", firstParameter.Name)
+	resp, _, err := TOSession.GetParameters(params, nil)
+	if err != nil {
+		t.Errorf("cannot GET Parameter by name: %v - %v", firstParameter.Name, err)
+	}
+	remoteParameter := resp[0]
+	//Parameters can be updated with empty value, so no error while updating
+	remoteParameter.Value = ""
+
+	var alert tc.Alerts
+	alert, reqInf, err := TOSession.UpdateParameter(remoteParameter.ID, remoteParameter, nil)
+	if err != nil {
+		t.Errorf("cannot UPDATE Parameter by id: %v - %v", err, alert)
+	}
+	if reqInf.StatusCode != http.StatusOK {
+		t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+	}
+}
+
+func UpdateParametersInvalidName(t *testing.T) {
+	firstParameter := testData.Parameters[0]
```

Comment:
This will segfault if the testing data is in a bad state

Reply:
Added null check for getting values.